File: src/main/java/me/leodev/trollingfreedomreborn/commands/AnvilDrop.java
package me.leodev.trollingfreedomreborn.commands;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;

public class AnvilDrop implements Listener {

    public static void Anvil(final Player p) {
        // 1. Capture the task ID (int)
        int taskId = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, new Runnable() {
            public void run() {
                if (p.isOnline()) {
                    Location loc1 = p.getLocation().add(p.getLocation().getDirection());
                    Block block1 = p.getWorld().getBlockAt(loc1.add(0.0D, 20.0D, 0.0D)); //MIDDLE
                    block1.setType(Material.DAMAGED_ANVIL);
                }
            }
        }, 40L, 10L);

        // 2. Register it with the "Anvil Drop" key
        // Make sure this name matches the item name in your GUI
        Core.instance.addTask(p, "anvildrop", taskId);
    }

    // You can remove UnAnvil(Plugin plugin) entirely now because
    // UnTroll.java will handle cancellation via the taskId map!
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/classics/AnvilDrop.kt
package com.leomadrassi.trollingfreedomreborn.trolls.classics

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.Listener

class AnvilDrop : Listener {

    companion object {
        fun Anvil(p: Player) {
            val taskId = Bukkit.getScheduler().scheduleSyncRepeatingTask(Core.instance, Runnable {
                if (p.isOnline) {
                    val loc1 = p.location.add(p.location.direction)
                    val block1 = p.world.getBlockAt(loc1.add(0.0, 20.0, 0.0))
                    block1.setType(Material.DAMAGED_ANVIL)
                }
            }, 40L, 10L)
            Core.instance.addTask(p, "anvildrop", taskId)
        }
    }
}
